pallets/identity/src/lib.rs

```diff
@@ -58,6 +71,19 @@ pub mod pallet {
         #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
         /// Create a new identity owned by origin.
         pub fn create_identity(origin: OriginFor<T>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let total_ids: u32 = <IdentityNumber<T>>::get();
+            let new_id: u32 = total_ids.checked_add(1).ok_or(Error::<T>::StorageOverflow)?;
+
+            <IdentityList<T>>::try_mutate(&who, |ids| -> DispatchResult {
```

Comment:
if StorageOverflow occurs, it will return without affecting storage + if try mutate fails, the `?` on line 82 should cause a return without changing state

Reply:
I think I may have just straight-up forgotten how `?` works for a second there. I'll chalk it up to not using Rust much for a while. 😂 Gave us a really good chance to talk out some improvements to both pallets, though.